crypto/src/proofs/transparent.rs

```diff
@@ -418,57 +436,109 @@ impl SwapClaimProof {
     /// * value commitment of the note to be spent,
     /// * nullifier of the note to be spent,
     /// * the randomized verification spend key,
+    /// * the pre-paid fee amount for the swap,
     pub fn verify(
         &self,
-        anchor: tct::builder::block::Root,
+        anchor: tct::Root,
         // Value commitment to the fees for the swap claim
         value_commitment: value::Commitment,
         nullifier: Nullifier,
         clearing_price_1: u64,
         clearing_price_2: u64,
```

Comment:
Rather than taking `clearing_price_i`, we'll want the proof verification to take a `BatchSwapOutputData`, which will let us compute the pro rata share of the output amount exactly.

Reply:
(This also means we don't have to add an explicit `success` bool below)